spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -1033,54 +1033,55 @@ def _construct_base_spark_application(
 
         return spark_app
 
-    # None of these abstract methods actually need to be implemented because we will be calling into the proper
+    # None of these abstract methods actually need to do anything because we will be calling into the proper
     #  un/parsed SparkApplication loader based on the underlying data, and those loaders have these methods
-    #  implemented already
+    #  implemented already. If some class sub-classes this one, then these methods will just echo back the
+    #  spark_app that we already constructed
```

Comment:
Hopefully this is self-explanatory, but if a subclass of this class calls `construct_spark_application()`, then these `compute_*` methods will get called. They just need to echo back the `spark_app` for everything to work as expected, whereas the prior `pass` implementation meant that it would return `None` (would be nicer if an exception was raised by Python, but alas...)